rosbag2/src/rosbag2/rosbag2.cpp

```diff
@@ -27,59 +29,136 @@
 #include "rosbag2_storage/storage_interfaces/read_only_interface.hpp"
 #include "rosbag2_storage/storage_interfaces/read_write_interface.hpp"
 #include "rosbag2_storage/storage_factory.hpp"
+#include "generic_subscription.hpp"
+#include "rosbag2_node.hpp"
+#include "typesupport_helpers.hpp"
 
 namespace rosbag2
 {
 
 const char * ROS_PACKAGE_NAME = "rosbag2";
 
+std::string Rosbag2::get_topic_type(
+  const std::string & topic_name, const std::shared_ptr<rclcpp::Node> & node)
+{
+  // TODO(Martin-Idel-SI): This is a short sleep to allow the node some time to discover the topic
+  // This should be replaced by an auto-discovery system in the future
```

Comment:
Here are my problems with this approach: 
- How do you plan to support subscribing to new topics which are not available at startup time? If I say `rosbag2 record -all` I would eventually expect also new topics that are created after startup to be recorded. In that case, this will have to be done in our C++ loops, so I don't think it's a good idea to do this in the python code.
- I thought that the python code would really only be a python wrapper. But if you get topics and types in python, then you'd have further functionality there. I think it would be cleaner to have all functionality in C++

Reply:
I see your point. I believe we could have an overload for record (for a later time):
* `record(const std::string & topic_name, ...)`
* `record(const std::string & topic_name, const std::string & topic_type)`

The reason I am bringing up the python interface is, because it's easier to load the type information from python directly.

The thing with recording new topics is tricky though. This has to be somehow specified how often we look for new topics then. I believe in the current ROS 1 implementation, only the topics available at startup time are recorded. Otherwise I would expect asking for all available topics after every `spin`.